onenote/endpoint.bal

```diff
@@ -0,0 +1,274 @@
+            path = path + QUESTION_MARK + oDataQuery;
+        }
+        http:Response response = check self.httpClient->get(path);
+        return getNotebookArray(response);
+    }
+
+    # Gets a notebook.
+    # 
+    # + notebookId - Notebook Id
+    # + return - Notebook if successful or else an error
+    @display {label: "Get Notebook"}
+    remote isolated function getNotebook(@display {label: "Notebook Id"} string notebookId) returns Notebook|error {
+        string path = NOTEBOOKS_PATH + SLASH + notebookId;
+        return check self.httpClient->get(path, targetType = Notebook);
+    }
+
+    # Lists recent notebooks.
+    # 
+    # + return - RecentNotebook[] if successful or else an error
+    @display {label: "Get Recent Notebooks"}
+    remote isolated function getRecentNotebooks() returns RecentNotebook[]|error {
+        string path = NOTEBOOKS_PATH + "/getRecentNotebooks(includePersonalNotebooks=true)";
+        http:Response response = check self.httpClient->get(path);
+        return getRecentNotebookArray(response);
+    }
+
+    // Sections
+
+    # Lists sections.
+    # 
+    # + notebookId - Notebook Id
+    # + oDataQuery - Optional. OData Query. Example: "$top=2&$count=true". 
+    # For details, refer https://docs.microsoft.com/en-us/graph/query-parameters#odata-system-query-options
+    # + return - Section[] if successful or else an error
+    @display {label: "List Sections"}
+    remote isolated function listSections(@display {label: "Notebook Id"} string notebookId, 
+                                          @display {label: "OData Query"} string? oDataQuery = ()) returns 
+                                          Section[]|error {
+        string path = NOTEBOOKS_PATH + SLASH + notebookId + SECTIONS;
+        if (oDataQuery is string) {
+            path = path + QUESTION_MARK + oDataQuery;
+        }
+        http:Response response = check self.httpClient->get(path);
+        return getSectionArray(response);
+    }
+
+    # Gets a section.
+    # 
+    # + sectionId - Section Id
+    # + return - Section if successful or else an error
+    @display {label: "Get Section"}
+    remote isolated function getSection(@display {label: "Section Id"} string sectionId) returns Section|error {
+        string path = SECTIONS_PATH + sectionId;
+        return check self.httpClient->get(path, targetType = Section);
+    }
+
+    # Creates a section.
+    # 
+    # + notebookId - Notebook Id
+    # + sectionName - Display name for the section
+    # + return - Section if successful or else an error
+    @display {label: "Create Section"}
+    remote isolated function createSection(@display {label: "Notebook Id"} string notebookId, 
+                                           @display {label: "Section Name"} string sectionName) returns Section|error {
+        string path = NOTEBOOKS_PATH + SLASH + notebookId + SECTIONS;
+        http:Response response = check self.httpClient->post(path, {displayName: sectionName});
```

Comment:
Can't we use `targetType`?

Reply:
Possible.
But i have already implemented a function called `handleResponse` and checking the statusCode and handling the mapping.

Reply:
Why do we need to handle this with extra work?

Reply:
Agreed.
Done the changes using targetType